pandas/io/pytables.py

```diff
@@ -1860,7 +1860,9 @@ def set_atom_string(self, block, block_items, existing_col, min_itemsize,
                     )
 
         # itemsize is the maximum length of a string (along any dimension)
-        itemsize = lib.max_len_string_array(com._ensure_object(data.ravel()))
+        data_converted = _convert_string_array(data, encoding)
+        itemsize = lib.max_len_string_array(
```

Comment:
in fact you don't need to do this, simply ready the computed itemsize from the dtype, e.g. ``itemsize = data_converted.itemsize``